src/ngraph/ngraph_nnvm_ops.cc

```diff
@@ -144,8 +158,17 @@ bool check_zero_grad(const std::shared_ptr<Graph> &graph) {
   auto size = graph->nodes_.size();
   if ((size < 1) || (graph->nodes_[size - 1]->type_ != NodeType::kOp))
     return false;
-  if (ops_no_head_grad.count(graph->nodes_[size - 1]->operation_)) return true;
-  return false;
+  bool zero_grad = true;
+  // if all of the outputs of the graph don't need gradient calculation,
+  // don't autodiff this graph. Otherwise, do.
+  for (auto node : graph->outputs_) {
+    if (ops_no_head_grad.count(node->operation_)) {
+      zero_grad = zero_grad && true;
```

Comment:
The new comment above is great - glad you added it.

But I was actually suggesting a comment regarding this line:
``` c++
zero_grad = zero_grad && true;
```

It's not obvious to my what you use a tautology on the rhs.

Reply:
Because I was drunk at the time, maybe?